src/common/numa.h

```diff
@@ -3,6 +3,13 @@
 
 #pragma once
 
+#if defined(__FreeBSD__)
+#include <sys/param.h>
+#include <sys/cpuset.h>
+#define cpu_set_t cpuset_t
```

Comment:
might want to use `typedef cpuset_t cpu_set_t` or `using cpu_set_t = cpuset_t` instead of using macro.

Reply:
also, might want to put
```c++
#if defined(__FreeBSD__)
...
#elif defined(__linux__)
#include <sched.h>
#endif 
```

Reply:
@tchaikov 
In all fairness `cpuset_t` is not equivalent to `cpu_set_t` in regards of what the numa routines use. 
the cpu sets are part of a larger structure in the the numa structure.
But as long as we are doing the shims it'll work, however when working on the implementation it might require a bit more work.

Reply:
@tchaikov 
Why would I want to remove the `sched.h`??
FreeBSD does have it, and they seem to be doing the same functions.

Reply:
> In all fairness `cpuset_t` is not equivalent to `cpu_set_t` in regards of what the numa routines use. 

then, please do not `#define` it. it's just misleading and wrong.

> FreeBSD does have it, and they seem to be doing the same functions.

could you enlighten me the reason why you include it? are you including it any type or function declaration offered by this header?

Reply:
@tchaikov 
Let me refrase this: the types are actually equal....
But to start implementing the shims, it will require some more structure, since the the cpuset is part of a structure that is consumed by the scheduling routines on FreeBSD which most only has the POSIX stuff.
But I'll give it some more consideration.

Reply:
again, i don't think we should include `<sched.h>` on FreeBSD for no reason.

Reply:
@tchaikov 
No problem, I'll take it out.